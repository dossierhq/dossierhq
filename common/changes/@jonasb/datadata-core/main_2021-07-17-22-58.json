{
  "changes": [
    {
      "packageName": "@jonasb/datadata-core",
      "comment": "Add OkResult.map() and ErrorType.isErrorType()",
      "type": "none"
    }
  ],
  "packageName": "@jonasb/datadata-core",
  "email": "jonas.b@gmail.com"
}