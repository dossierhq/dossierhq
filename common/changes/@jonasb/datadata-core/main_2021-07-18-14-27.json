{
  "changes": [
    {
      "packageName": "@jonasb/datadata-core",
      "comment": "Configure importHelpers for ts, and add tslib dependency",
      "type": "none"
    }
  ],
  "packageName": "@jonasb/datadata-core",
  "email": "jonas.b@gmail.com"
}