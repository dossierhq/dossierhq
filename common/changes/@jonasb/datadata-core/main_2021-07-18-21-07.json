{
  "changes": [
    {
      "packageName": "@jonasb/datadata-core",
      "comment": "Export transpiled JavaScript",
      "type": "none"
    }
  ],
  "packageName": "@jonasb/datadata-core",
  "email": "jonas.b@gmail.com"
}