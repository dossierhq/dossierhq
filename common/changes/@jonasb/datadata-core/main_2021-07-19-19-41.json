{
  "changes": [
    {
      "packageName": "@jonasb/datadata-core",
      "comment": "Fix export configuration",
      "type": "none"
    }
  ],
  "packageName": "@jonasb/datadata-core",
  "email": "jonas.b@gmail.com"
}