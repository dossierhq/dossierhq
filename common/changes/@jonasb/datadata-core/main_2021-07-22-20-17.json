{
  "changes": [
    {
      "packageName": "@jonasb/datadata-core",
      "comment": "Allow string in schema field specification",
      "type": "none"
    }
  ],
  "packageName": "@jonasb/datadata-core",
  "email": "jonas.b@gmail.com"
}