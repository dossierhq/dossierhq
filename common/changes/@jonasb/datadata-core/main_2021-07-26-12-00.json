{
  "changes": [
    {
      "packageName": "@jonasb/datadata-core",
      "comment": "Enable easier JSON serialization of adminClient",
      "type": "none"
    }
  ],
  "packageName": "@jonasb/datadata-core",
  "email": "jonas.b@gmail.com"
}