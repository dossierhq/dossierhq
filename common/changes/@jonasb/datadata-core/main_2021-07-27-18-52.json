{
  "changes": [
    {
      "packageName": "@jonasb/datadata-core",
      "comment": "Support boolean type",
      "type": "none"
    }
  ],
  "packageName": "@jonasb/datadata-core",
  "email": "jonas.b@gmail.com"
}