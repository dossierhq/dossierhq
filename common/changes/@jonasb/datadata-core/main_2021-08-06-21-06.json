{
  "changes": [
    {
      "packageName": "@jonasb/datadata-core",
      "comment": "upsertEntity() support in AdminClient",
      "type": "none"
    }
  ],
  "packageName": "@jonasb/datadata-core",
  "email": "jonas.b@gmail.com"
}