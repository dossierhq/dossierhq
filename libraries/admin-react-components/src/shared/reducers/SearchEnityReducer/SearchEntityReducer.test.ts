import type {
  Connection,
  Edge,
  EntitySamplingPayload,
  ErrorType,
  PublishedEntity,
} from '@jonasb/datadata-core';
import { AdminQueryOrder, ok } from '@jonasb/datadata-core';
import { Temporal } from '@js-temporal/polyfill';
import type { SearchEntityState, SearchEntityStateAction } from './SearchEntityReducer';
import {
  initializeSearchEntityState,
  reduceSearchEntityState,
  SearchEntityStateActions,
} from './SearchEntityReducer';

function createPublishedEntity(id: string): PublishedEntity {
  return {
    id,
    info: {
      name: `Entity ${id}`,
      type: 'TitleOnly',
      authKey: 'none',
      createdAt: Temporal.Instant.from('2022-03-19T07:51:25.56Z'),
    },
    fields: { title: `Title ${id}` },
  };
}

function createPublishedEntityConnection(
  ids: string[],
  { hasPreviousPage, hasNextPage }: { hasPreviousPage: boolean; hasNextPage: boolean }
): Connection<Edge<PublishedEntity, ErrorType>> {
  return {
    pageInfo: {
      hasPreviousPage,
      hasNextPage,
      startCursor: `cursor-${ids[0]}`,
      endCursor: `cursor-${ids[ids.length - 1]}`,
    },
    edges: ids.map((id) => ({
      cursor: `cursor-${id}`,
      node: ok(createPublishedEntity(id)),
    })),
  };
}

function createPublishedEntitySamplingPayload(
  ids: string[],
  { seed, totalCount }: { seed: number; totalCount: number }
): EntitySamplingPayload<PublishedEntity> {
  return {
    seed,
    totalCount,
    items: ids.map(createPublishedEntity),
  };
}

function reduceSearchEntityStateActions(
  state: SearchEntityState,
  ...actions: SearchEntityStateAction[]
) {
  let newState = state;
  for (const action of actions) {
    newState = reduceSearchEntityState(newState, action);
  }
  return newState;
}

describe('initializeSearchEntityState', () => {
  test('default', () => {
    expect(initializeSearchEntityState([])).toMatchInlineSnapshot(`
      Object {
        "connection": undefined,
        "connectionError": undefined,
        "entities": Array [],
        "entitySamples": undefined,
        "entitySamplesError": undefined,
        "paging": Object {},
        "query": Object {
          "order": "name",
        },
        "requestedCount": 25,
        "sampling": undefined,
        "text": "",
        "totalCount": null,
      }
    `);
  });
});

describe('SearchEntityStateActions.SetQuery', () => {
  test('Resets sampling if setting order', () => {
    const state = reduceSearchEntityState(
      initializeSearchEntityState([new SearchEntityStateActions.SetSampling({}, true)]),
      new SearchEntityStateActions.SetQuery(
        { order: AdminQueryOrder.updatedAt },
        { partial: true, resetPagingIfModifying: true }
      )
    );
    expect(state.query).toEqual({ order: AdminQueryOrder.updatedAt });
    expect(state.paging).toEqual({});
    expect(state.sampling).toBeUndefined();
  });

  test('Resets paging (when configured)', () => {
    const state = reduceSearchEntityState(
      initializeSearchEntityState([
        new SearchEntityStateActions.SetPaging({ after: 'cursor', first: 10 }),
      ]),
      new SearchEntityStateActions.SetQuery(
        { order: AdminQueryOrder.updatedAt },
        { partial: true, resetPagingIfModifying: true }
      )
    );
    expect(state.query).toEqual({ order: AdminQueryOrder.updatedAt });
    expect(state.paging).toEqual({});
    expect(state.sampling).toBeUndefined();
  });
});

describe('SearchEntityStateActions.SetPaging', () => {
  test('Empty paging', () => {
    const state = reduceSearchEntityState(
      initializeSearchEntityState([]),
      new SearchEntityStateActions.SetPaging({})
    );
    expect(state.paging).toEqual({});
    expect(state.sampling).toBeUndefined();
  });

  test('Resets sampling', () => {
    const state = reduceSearchEntityState(
      initializeSearchEntityState([new SearchEntityStateActions.SetSampling({}, true)]),
      new SearchEntityStateActions.SetPaging({})
    );
    expect(state.paging).toEqual({});
    expect(state.sampling).toBeUndefined();
  });
});

describe('SearchEntityStateActions.SetSampling', () => {
  test('Empty sampling', () => {
    const state = reduceSearchEntityState(
      initializeSearchEntityState([]),
      new SearchEntityStateActions.SetSampling({}, true)
    );
    expect(state.sampling).toEqual({ seed: expect.any(Number) });
    expect(state.paging).toBeUndefined();
  });

  test('Resets paging', () => {
    const state = reduceSearchEntityState(
      initializeSearchEntityState([new SearchEntityStateActions.SetPaging({ first: 200 })]),
      new SearchEntityStateActions.SetSampling({}, true)
    );
    expect(state.sampling).toEqual({ seed: expect.any(Number) });
    expect(state.paging).toBeUndefined();
  });

  test('Resets order', () => {
    const state = reduceSearchEntityState(
      initializeSearchEntityState([
        new SearchEntityStateActions.SetQuery(
          { order: AdminQueryOrder.updatedAt },
          { partial: true, resetPagingIfModifying: true }
        ),
      ]),
      new SearchEntityStateActions.SetSampling({}, true)
    );
    expect(state.sampling).toEqual({ seed: expect.any(Number) });
    expect(state.query).toEqual({});
  });

  test('Partial count, keeps seed', () => {
    const state = reduceSearchEntityState(
      initializeSearchEntityState([new SearchEntityStateActions.SetSampling({ seed: 123 }, false)]),
      new SearchEntityStateActions.SetSampling({ count: 100 }, true)
    );
    expect(state.sampling).toEqual({ seed: 123, count: 100 });
    expect(state.requestedCount).toBe(100);
  });
});

describe('SearchEntityState scenarios', () => {
  test('Next page -> loading -> loaded', () => {
    const initialState = initializeSearchEntityState([
      new SearchEntityStateActions.SetPaging({ first: 1 }),
      new SearchEntityStateActions.UpdateSearchResult(
        createPublishedEntityConnection(['1'], { hasPreviousPage: false, hasNextPage: true }),
        undefined
      ),
      new SearchEntityStateActions.UpdateTotalCount(2),
    ]);
    expect(initialState).toMatchSnapshot('1 - initial');

    const loadingState = reduceSearchEntityStateActions(
      initialState,
      new SearchEntityStateActions.SetPaging({ after: 'cursor-1' }),
      new SearchEntityStateActions.UpdateSearchResult(undefined, undefined)
    );
    expect(loadingState).toMatchSnapshot('2 - loading');

    const loadedState = reduceSearchEntityState(
      loadingState,
      new SearchEntityStateActions.UpdateSearchResult(
        createPublishedEntityConnection(['2'], { hasPreviousPage: true, hasNextPage: false }),
        undefined
      )
    );
    expect(loadedState).toMatchSnapshot('3 - loaded');
  });

  test('Set sampling -> loading -> loaded', () => {
    const initialState = initializeSearchEntityState([
      new SearchEntityStateActions.SetPaging({ first: 1 }),
      new SearchEntityStateActions.UpdateSearchResult(
        createPublishedEntityConnection(['1'], { hasPreviousPage: false, hasNextPage: true }),
        undefined
      ),
      new SearchEntityStateActions.UpdateTotalCount(2),
      new SearchEntityStateActions.UpdateSampleResult(undefined, undefined),
    ]);
    expect(initialState).toMatchSnapshot('1 - initial');

    const loadingState = reduceSearchEntityStateActions(
      initialState,
      new SearchEntityStateActions.SetSampling({ count: 1, seed: 123 }, false),
      new SearchEntityStateActions.UpdateSearchResult(undefined, undefined),
      new SearchEntityStateActions.UpdateTotalCount(null)
    );
    expect(loadingState).toMatchSnapshot('2 - loading');

    const loadedState = reduceSearchEntityState(
      loadingState,
      new SearchEntityStateActions.UpdateSampleResult(
        createPublishedEntitySamplingPayload(['2'], { seed: 123, totalCount: 1 }),
        undefined
      )
    );
    expect(loadedState).toMatchSnapshot('3 - loaded');
  });
});
