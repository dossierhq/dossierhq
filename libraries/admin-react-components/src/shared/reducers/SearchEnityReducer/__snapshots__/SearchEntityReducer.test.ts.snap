// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchEntityState scenarios From sampling -> set paging -> loading -> loaded: 1 - initial 1`] = `
Object {
  "connection": undefined,
  "connectionError": undefined,
  "entities": Array [
    OkResult {
      "value": Object {
        "fields": Object {
          "title": "Title 1",
        },
        "id": "1",
        "info": Object {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 1",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 1,
  "entitySamples": Object {
    "items": Array [
      Object {
        "fields": Object {
          "title": "Title 1",
        },
        "id": "1",
        "info": Object {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 1",
          "type": "TitleOnly",
        },
      },
    ],
    "seed": 123,
    "totalCount": 1,
  },
  "entitySamplesError": undefined,
  "loadingState": "",
  "paging": undefined,
  "query": Object {},
  "requestedCount": 1,
  "sampling": Object {
    "count": 1,
    "seed": 123,
  },
  "text": "",
  "totalCount": null,
}
`;

exports[`SearchEntityState scenarios From sampling -> set paging -> loading -> loaded: 2 - loading 1`] = `
Object {
  "connection": undefined,
  "connectionError": undefined,
  "entities": Array [
    OkResult {
      "value": Object {
        "fields": Object {
          "title": "Title 1",
        },
        "id": "1",
        "info": Object {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 1",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 1,
  "entitySamples": undefined,
  "entitySamplesError": undefined,
  "loadingState": "first-page",
  "paging": Object {
    "first": 1,
  },
  "query": Object {},
  "requestedCount": 1,
  "sampling": undefined,
  "text": "",
  "totalCount": null,
}
`;

exports[`SearchEntityState scenarios From sampling -> set paging -> loading -> loaded: 3 - loaded 1`] = `
Object {
  "connection": Object {
    "edges": Array [
      Object {
        "cursor": "cursor-2",
        "node": OkResult {
          "value": Object {
            "fields": Object {
              "title": "Title 2",
            },
            "id": "2",
            "info": Object {
              "authKey": "none",
              "createdAt": "2022-03-19T07:51:25.56Z",
              "name": "Entity 2",
              "type": "TitleOnly",
            },
          },
        },
      },
    ],
    "pageInfo": Object {
      "endCursor": "cursor-2",
      "hasNextPage": true,
      "hasPreviousPage": false,
      "startCursor": "cursor-2",
    },
  },
  "connectionError": undefined,
  "entities": Array [
    OkResult {
      "value": Object {
        "fields": Object {
          "title": "Title 2",
        },
        "id": "2",
        "info": Object {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 2",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 2,
  "entitySamples": undefined,
  "entitySamplesError": undefined,
  "loadingState": "",
  "paging": Object {
    "first": 1,
  },
  "query": Object {},
  "requestedCount": 1,
  "sampling": undefined,
  "text": "",
  "totalCount": 2,
}
`;

exports[`SearchEntityState scenarios From searching -> set sampling -> loading -> loaded: 1 - initial 1`] = `
Object {
  "connection": Object {
    "edges": Array [
      Object {
        "cursor": "cursor-1",
        "node": OkResult {
          "value": Object {
            "fields": Object {
              "title": "Title 1",
            },
            "id": "1",
            "info": Object {
              "authKey": "none",
              "createdAt": "2022-03-19T07:51:25.56Z",
              "name": "Entity 1",
              "type": "TitleOnly",
            },
          },
        },
      },
    ],
    "pageInfo": Object {
      "endCursor": "cursor-1",
      "hasNextPage": true,
      "hasPreviousPage": false,
      "startCursor": "cursor-1",
    },
  },
  "connectionError": undefined,
  "entities": Array [
    OkResult {
      "value": Object {
        "fields": Object {
          "title": "Title 1",
        },
        "id": "1",
        "info": Object {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 1",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 0,
  "entitySamples": undefined,
  "entitySamplesError": undefined,
  "loadingState": "",
  "paging": Object {
    "first": 1,
  },
  "query": Object {
    "order": "name",
  },
  "requestedCount": 1,
  "sampling": undefined,
  "text": "",
  "totalCount": 2,
}
`;

exports[`SearchEntityState scenarios From searching -> set sampling -> loading -> loaded: 2 - loading 1`] = `
Object {
  "connection": undefined,
  "connectionError": undefined,
  "entities": Array [
    OkResult {
      "value": Object {
        "fields": Object {
          "title": "Title 1",
        },
        "id": "1",
        "info": Object {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 1",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 0,
  "entitySamples": undefined,
  "entitySamplesError": undefined,
  "loadingState": "sample",
  "paging": undefined,
  "query": Object {},
  "requestedCount": 1,
  "sampling": Object {
    "count": 1,
    "seed": 123,
  },
  "text": "",
  "totalCount": null,
}
`;

exports[`SearchEntityState scenarios From searching -> set sampling -> loading -> loaded: 3 - loaded 1`] = `
Object {
  "connection": undefined,
  "connectionError": undefined,
  "entities": Array [
    OkResult {
      "value": Object {
        "fields": Object {
          "title": "Title 2",
        },
        "id": "2",
        "info": Object {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 2",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 1,
  "entitySamples": Object {
    "items": Array [
      Object {
        "fields": Object {
          "title": "Title 2",
        },
        "id": "2",
        "info": Object {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 2",
          "type": "TitleOnly",
        },
      },
    ],
    "seed": 123,
    "totalCount": 1,
  },
  "entitySamplesError": undefined,
  "loadingState": "",
  "paging": undefined,
  "query": Object {},
  "requestedCount": 1,
  "sampling": Object {
    "count": 1,
    "seed": 123,
  },
  "text": "",
  "totalCount": null,
}
`;

exports[`SearchEntityState scenarios Next page -> loading -> loaded: 1 - initial 1`] = `
Object {
  "connection": Object {
    "edges": Array [
      Object {
        "cursor": "cursor-1",
        "node": OkResult {
          "value": Object {
            "fields": Object {
              "title": "Title 1",
            },
            "id": "1",
            "info": Object {
              "authKey": "none",
              "createdAt": "2022-03-19T07:51:25.56Z",
              "name": "Entity 1",
              "type": "TitleOnly",
            },
          },
        },
      },
    ],
    "pageInfo": Object {
      "endCursor": "cursor-1",
      "hasNextPage": true,
      "hasPreviousPage": false,
      "startCursor": "cursor-1",
    },
  },
  "connectionError": undefined,
  "entities": Array [
    OkResult {
      "value": Object {
        "fields": Object {
          "title": "Title 1",
        },
        "id": "1",
        "info": Object {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 1",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 0,
  "entitySamples": undefined,
  "entitySamplesError": undefined,
  "loadingState": "",
  "paging": Object {
    "first": 1,
  },
  "query": Object {
    "order": "name",
  },
  "requestedCount": 1,
  "sampling": undefined,
  "text": "",
  "totalCount": 2,
}
`;

exports[`SearchEntityState scenarios Next page -> loading -> loaded: 2 - loading 1`] = `
Object {
  "connection": undefined,
  "connectionError": undefined,
  "entities": Array [
    OkResult {
      "value": Object {
        "fields": Object {
          "title": "Title 1",
        },
        "id": "1",
        "info": Object {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 1",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 0,
  "entitySamples": undefined,
  "entitySamplesError": undefined,
  "loadingState": "next-page",
  "paging": Object {
    "after": "cursor-1",
  },
  "query": Object {
    "order": "name",
  },
  "requestedCount": 1,
  "sampling": undefined,
  "text": "",
  "totalCount": 2,
}
`;

exports[`SearchEntityState scenarios Next page -> loading -> loaded: 3 - loaded 1`] = `
Object {
  "connection": Object {
    "edges": Array [
      Object {
        "cursor": "cursor-2",
        "node": OkResult {
          "value": Object {
            "fields": Object {
              "title": "Title 2",
            },
            "id": "2",
            "info": Object {
              "authKey": "none",
              "createdAt": "2022-03-19T07:51:25.56Z",
              "name": "Entity 2",
              "type": "TitleOnly",
            },
          },
        },
      },
    ],
    "pageInfo": Object {
      "endCursor": "cursor-2",
      "hasNextPage": false,
      "hasPreviousPage": true,
      "startCursor": "cursor-2",
    },
  },
  "connectionError": undefined,
  "entities": Array [
    OkResult {
      "value": Object {
        "fields": Object {
          "title": "Title 2",
        },
        "id": "2",
        "info": Object {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 2",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 1,
  "entitySamples": undefined,
  "entitySamplesError": undefined,
  "loadingState": "",
  "paging": Object {
    "after": "cursor-1",
  },
  "query": Object {
    "order": "name",
  },
  "requestedCount": 1,
  "sampling": undefined,
  "text": "",
  "totalCount": 2,
}
`;

exports[`SearchEntityState scenarios Refresh: 1 - initial 1`] = `
Object {
  "connection": Object {
    "edges": Array [
      Object {
        "cursor": "cursor-1",
        "node": OkResult {
          "value": Object {
            "fields": Object {
              "title": "Title 1",
            },
            "id": "1",
            "info": Object {
              "authKey": "none",
              "createdAt": "2022-03-19T07:51:25.56Z",
              "name": "Entity 1",
              "type": "TitleOnly",
            },
          },
        },
      },
    ],
    "pageInfo": Object {
      "endCursor": "cursor-1",
      "hasNextPage": true,
      "hasPreviousPage": false,
      "startCursor": "cursor-1",
    },
  },
  "connectionError": undefined,
  "entities": Array [
    OkResult {
      "value": Object {
        "fields": Object {
          "title": "Title 1",
        },
        "id": "1",
        "info": Object {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 1",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 0,
  "entitySamples": undefined,
  "entitySamplesError": undefined,
  "loadingState": "",
  "paging": Object {
    "first": 1,
  },
  "query": Object {
    "order": "name",
  },
  "requestedCount": 1,
  "sampling": undefined,
  "text": "",
  "totalCount": 2,
}
`;

exports[`SearchEntityState scenarios Refresh: 2 - loaded 1`] = `
Object {
  "connection": Object {
    "edges": Array [
      Object {
        "cursor": "cursor-2",
        "node": OkResult {
          "value": Object {
            "fields": Object {
              "title": "Title 2",
            },
            "id": "2",
            "info": Object {
              "authKey": "none",
              "createdAt": "2022-03-19T07:51:25.56Z",
              "name": "Entity 2",
              "type": "TitleOnly",
            },
          },
        },
      },
    ],
    "pageInfo": Object {
      "endCursor": "cursor-2",
      "hasNextPage": true,
      "hasPreviousPage": false,
      "startCursor": "cursor-2",
    },
  },
  "connectionError": undefined,
  "entities": Array [
    OkResult {
      "value": Object {
        "fields": Object {
          "title": "Title 2",
        },
        "id": "2",
        "info": Object {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 2",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 0,
  "entitySamples": undefined,
  "entitySamplesError": undefined,
  "loadingState": "",
  "paging": Object {
    "first": 1,
  },
  "query": Object {
    "order": "name",
  },
  "requestedCount": 1,
  "sampling": undefined,
  "text": "",
  "totalCount": 2,
}
`;

exports[`SearchEntityState scenarios Sampling -> change seed -> loading -> loaded: 1 - initial 1`] = `
Object {
  "connection": undefined,
  "connectionError": undefined,
  "entities": Array [
    OkResult {
      "value": Object {
        "fields": Object {
          "title": "Title 1",
        },
        "id": "1",
        "info": Object {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 1",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 1,
  "entitySamples": Object {
    "items": Array [
      Object {
        "fields": Object {
          "title": "Title 1",
        },
        "id": "1",
        "info": Object {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 1",
          "type": "TitleOnly",
        },
      },
    ],
    "seed": 123,
    "totalCount": 2,
  },
  "entitySamplesError": undefined,
  "loadingState": "",
  "paging": undefined,
  "query": Object {},
  "requestedCount": 1,
  "sampling": Object {
    "count": 1,
    "seed": 123,
  },
  "text": "",
  "totalCount": null,
}
`;

exports[`SearchEntityState scenarios Sampling -> change seed -> loading -> loaded: 2 - loading 1`] = `
Object {
  "connection": undefined,
  "connectionError": undefined,
  "entities": Array [
    OkResult {
      "value": Object {
        "fields": Object {
          "title": "Title 1",
        },
        "id": "1",
        "info": Object {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 1",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 1,
  "entitySamples": undefined,
  "entitySamplesError": undefined,
  "loadingState": "sample",
  "paging": undefined,
  "query": Object {},
  "requestedCount": 1,
  "sampling": Object {
    "count": 1,
    "seed": 456,
  },
  "text": "",
  "totalCount": null,
}
`;

exports[`SearchEntityState scenarios Sampling -> change seed -> loading -> loaded: 3 - loaded 1`] = `
Object {
  "connection": undefined,
  "connectionError": undefined,
  "entities": Array [
    OkResult {
      "value": Object {
        "fields": Object {
          "title": "Title 2",
        },
        "id": "2",
        "info": Object {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 2",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 2,
  "entitySamples": Object {
    "items": Array [
      Object {
        "fields": Object {
          "title": "Title 2",
        },
        "id": "2",
        "info": Object {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 2",
          "type": "TitleOnly",
        },
      },
    ],
    "seed": 456,
    "totalCount": 2,
  },
  "entitySamplesError": undefined,
  "loadingState": "",
  "paging": undefined,
  "query": Object {},
  "requestedCount": 1,
  "sampling": Object {
    "count": 1,
    "seed": 456,
  },
  "text": "",
  "totalCount": null,
}
`;
